templates/build-dockerfiles/manifests.yaml

```diff
@@ -0,0 +1,214 @@
+            --redis-cache-addr={{inputs.parameters.redis-cache-address}} \
+            -t={{inputs.parameters.dockerfile-tag}} \
+            -f={{inputs.artifacts.git-repo.path}}/{{inputs.parameters.dockerfile-to-build}} \
+            {{inputs.parameters.dockerfile-context}}
+        volumeMounts:
+          - name: docker-registry-config
+            mountPath: /registry-config
+    - name: makisu
+      inputs:
+        parameters:
+        - name: dockerfile-to-build
+        - name: dockerfile-context
+        - name: dockerfile-tag
+        - name: redis-cache-address
+          value: "{{inputs.parameters.redis-cache-address}}"
+        artifacts:
+        - name: git-repo
+          path: "{{inputs.parameters.git-clone-artifact-path}}"
+      container:
+        image: gcr.io/uber-container-tools/makisu:latest
+        args: [
+          "build",
+          "--registry-config=/registry-config/registries-bdata-ml.yaml",
+          "--push=gcr.io",
+          "--modifyfs=true",
+          "--redis-cache-addr={{inputs.parameters.redis-cache-address}}",
+          "--compression=no",
+          "-t={{inputs.parameters.dockerfile-tag}}",
+          "-f={{inputs.artifacts.git-repo.path}}/{{inputs.parameters.dockerfile-to-build}}",
+          "{{inputs.parameters.dockerfile-context}}"
+        ]
+        workingDir: "{{inputs.artifacts.git-repo.path}}"
+        volumeMounts:
+          - name: docker-registry-config
+            mountPath: /registry-config
+    - name: kaniko
+      inputs:
+        parameters:
+        - name: dockerfile-to-build
+        - name: dockerfile-context
+        - name: dockerfile-tag
+        artifacts:
+        - name: git-repo
+          path: "{{inputs.parameters.git-clone-artifact-path}}"
+      container:
+        image: gcr.io/kaniko-project/executor:latest
+        args: [
+          "--cache=true",
+          "--destination={{inputs.parameters.dockerfile-tag}}",
+          "--dockerfile={{inputs.artifacts.git-repo.path}}/{{inputs.parameters.dockerfile-to-build}}",
+          "--context=dir://{{inputs.parameters.dockerfile-context}}"
+        ]
+        workingDir: "{{inputs.artifacts.git-repo.path}}"
+    - name: docker-in-docker
+      inputs:
+        parameters:
+        - name: dockerfile-to-build
+        - name: dockerfile-context
+        - name: dockerfile-tag
+        - name: project-shorthand
+        artifacts:
+        - name: git-repo
+          path: "{{inputs.parameters.git-clone-artifact-path}}"
+      script:
+        image: docker:19.03.8
+        imagePullPolicy: Always
+        command: [sh]
+        env:
+          - name: DOCKER_HOST
+            value: 127.0.0.1
+          - name: DOCKER_TLS
+            value: ""
+        workingDir: "{{inputs.artifacts.git-repo.path}}"
+        source: |
+          until docker ps; do
+            sleep 3
+          done
+          cat /var/secrets/google/npd/key.json | docker login -u _json_key --password-stdin https://gcr.io
```

Comment:
Similar to the comment above, I think this needs to support registries other than GCR